contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -2135,13 +2142,30 @@ exec_alter_role_cmd(char *query_str, RoleSpec *role)
  * Helper function to generate GRANT on SCHEMA subcommands.
  */
 static List
-*gen_grantschema_subcmds(const char *schema, const char *rolname, bool is_grant, bool with_grant_option, AclMode privilege)
+*gen_grantschema_subcmds(const char *schema, const char *rolname, bool is_grant, bool with_grant_option, AclMode privilege, bool is_create_schema)
 {
 	StringInfoData query;
 	List	   *stmt_list;
 	Node	   *stmt;
 	int			expected_stmts = 2;
 	int			i = 0;
+	bool		owner_other_than_dbo = false;
+	char 		*dbname = get_cur_db_name();
+	const char	*dbo_role = get_dbo_role_name(dbname);
+	const char	*db_owner_role = get_db_owner_name(dbname);
+	const char	*schema_owner = GetUserNameFromId(get_owner_of_schema(schema), false);
+
+	/*
+	 * Don't need multiple ALTER DEFAULT PRIVILEGE statements, if:
+	 * 1. It's a part of CREATE SCHEMA statement
+	 * 2. If the schema owner is not dbo
+	 * 3. If the schema owner is not db_owner
```

Comment:
Question: why is this needed only if schema owner is dbo or db_owner? Ideally if a user is granted SELECT on a particular schema then no matter who the owner is, that user should be able SELECT from any existing and new objects in that schema isn't it?

Reply:
We don't need more than 1 role for ALTER DEFAULT DEFAULT PRIVILEGES in case of schema owner being `dbo` (schema created by dbo user) or db_owner(schemas like master_dbo, tempdb_dbo etc which are owned by db_owner).